restcomm.android.client.sdk/src/main/java/org/mobicents/restcomm/android/client/sdk/RCClient.java

```diff
@@ -175,20 +174,20 @@ public static boolean isInitialized()
 
     /**
      * Create an initialize a new Device object
-     * @param parameters  Restcomm Client parameters
+     * @param profileConfig  Restcomm Client parameters
      * @param deviceListener  The listener for upcoming RCDevice events
      * @return The newly created RCDevice
      * @see RCDevice
      */
-    public static RCDevice createDevice(HashMap<String, Object> parameters, RCDeviceListener deviceListener)
+    public static RCDevice createDevice(RCDevice.SIPProfileConfig profileConfig, RCDeviceListener deviceListener)
```

Comment:
@mehturt: not sure it's a good idea to change the API at this point. People are already using it and we 'll break all those apps if we change that. If you want to add more options, I would suggest that you do so in the context of the existing data structure by adding new members in the HashMap